### PR TITLE
feat(2222, 2647): passion feed for gallery and landing page

### LINKED ISSUE #2222: Passion feed filter to gallery view v1.1 - Yulia
- [ ] add a switch to the main gallery view to switch to passion feed, default on. To be able to switch between ALL <> PASSION
- [ ] query
-- where a collector is a current owner but not the issuer
-- all nfts where the issuer is in the array from list 1

@vikiival will pass query shortly which should be put as resolver to https://github.com/kodadot/rubick

### LINKED ISSUE #2647: Passion filter on landing page
Once the user is logged in, show only 
- recent sales from collection users' wallets already bought
- new listings from the collection 

### Ref 
- #2226 
- #2223 
- #2222 
- #2620 
- #2622

### PR DESCRIPTION
**Pull request by KngZhi**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [x] Feature

### What's new?

- [x] PR closes #2222 
- [x] PR closes #2647
- feat: passion feed request
- feat: passion feed switch ui

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [x] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer?target=DaWoVNNhBTdqJ5iPN3UcaLeamkPpJeGWdrs8LhvX9TQbadc)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

https://user-images.githubusercontent.com/11361147/168478846-dde07e4a-2d8c-47e9-b559-ea3bbbb920ee.mov

### CODE REVIEW

**Comment by KngZhi** [comment]
You can check the effect when Rubick was deployed.

**Comment by yangwao** [comment]
> You can check the effect when Rubick was deployed.

let's a link which PR on Rubick?

**Comment by KngZhi** [comment]
> > You can check the effect when Rubick was deployed.
> 
> let's a link which PR on Rubick?

updated

**Comment by petersopko** [comment]
@KngZhi actually, I've found out what the problem is with the linked issue which you've been unassigned from (#2647) 
If you want to close multiple issues with one PR you need to use different format, i.e. `closes #1, closes #2`, otherwise bot doesn't know it's closing issue and not just linked issue 🚀  https://stackoverflow.com/questions/60027222/github-how-can-i-close-the-two-issues-with-commit-message

**Comment by KngZhi** [comment]
> @KngZhi actually, I've found out what the problem is with the linked issue which you've been unassigned from (#2647) If you want to close multiple issues with one PR you need to use different format, i.e. `closes #1, closes #2`, otherwise bot doesn't know it's closing issue and not just linked issue 🚀 https://stackoverflow.com/questions/60027222/github-how-can-i-close-the-two-issues-with-commit-message

@petersopko Oh, I see, learned, didn't realize the `closes` matters. Many thanks for the info and link!

File: components/landing/Landing.vue [hunk 1]
```diff
@@ -41,20 +41,46 @@
...
+    const {
+      data: { passionFeed },
+    } = await this.$apollo.query({
+      query: passionQuery,
+      client: 'subsquid',
+      variables: {
+        account: this.accountId,
+      },
+    })
+    return passionFeed.map((item) => item.id)
```
Comment (kkukelka) [inline components/landing/Landing.vue:82]:
```suggestion
    return passionFeed?.map(item => item.id) || []
```

File: components/rmrk/Gallery/Gallery.vue [hunk 2]
```diff
@@ -225,6 +251,19 @@ export default class Gallery extends mixins(PrefixMixin, InfiniteScrollMixin) {
...
+    const {
+      data: { passionFeed },
+    } = await this.$apollo.query({
+      query: passionQuery,
+      client: 'subsquid',
+      variables: {
+        account: this.accountId,
+      },
+    })
+    this.passionList = passionFeed.map((x) => x.id)
```
Comment (kkukelka) [inline components/rmrk/Gallery/Gallery.vue:264]:
```suggestion
    this.passionList = passionFeed?.map(x => x.id) || []
```

File: components/rmrk/Gallery/NewestList.vue [hunk 3]
```diff
@@ -43,7 +44,9 @@ const components = {
 @Component<NewestList>({
   components,
 })
-export default class NewestList extends mixins(PrefixMixin) {
+export default class NewestList extends mixins(PrefixMixin, AuthMixin) {
+  @Prop({ required: false, default: [''] }) passionList: string[]
```
Comment (kkukelka) [inline components/rmrk/Gallery/NewestList.vue:48]:
```suggestion
  @Prop({ type: Array, default: () => []}) passionList?: string[]
```

File: components/rmrk/Gallery/LatestSales.vue [hunk 4]
```diff
@@ -41,7 +42,9 @@ const components = {
 @Component<LatestSales>({
   components,
 })
-export default class LatestSales extends mixins(PrefixMixin) {
+export default class LatestSales extends mixins(PrefixMixin, AuthMixin) {
+  @Prop({ required: false, default: [''] }) passionList: string[]
```
Comment (kkukelka) [inline components/rmrk/Gallery/LatestSales.vue:46]:
```suggestion
  @Prop({ type: Array, default: () => [] }) passionList?: string[]
```

**Review by kkukelka — CHANGES_REQUESTED** [review]
sorry, i totally forgot about this PR. would these changes be ok with you?

**Comment by KngZhi** [comment]
> sorry, i totally forgot about this PR. would these changes be ok with you?

No worries :). Those are definitely good changes, thanks for the review, Kilian.

**Review by kkukelka — APPROVED** [review]
all good now!

**Comment by kkukelka** [comment]
pay 350

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $350 @ 67.34 USD/KSM ~ 5.198 $KSM
🧗 [DaWoVNNhBTdqJ5iPN3UcaLeamkPpJeGWdrs8LhvX9TQbadc](https://kusama.subscan.io/account/DaWoVNNhBTdqJ5iPN3UcaLeamkPpJeGWdrs8LhvX9TQbadc)
🔗 [0x3d58d29f06f6f569332069bbc94bb798382e4f43df30856c4c4f181d7e7a8e9d](https://kusama.subscan.io/extrinsic/0x3d58d29f06f6f569332069bbc94bb798382e4f43df30856c4c4f181d7e7a8e9d)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**